### PR TITLE
[WIP] Make BigBuf take dynamically the available space with a fixed (4K) stack

### PR DESCRIPTION
**Pull request by slurdge**
The work on the arm side is done.
We need an API call or some other way to forward the available space to the client.

### CODE REVIEW

File: common_arm/ldscript.common [hunk 1]
```diff
@@ -19,6 +21,7 @@ MEMORY
     bootphase2 : ORIGIN = 0x00100200, LENGTH = 0x2000 - 0x200    /* Main bootloader code, stored in Flash, executed from RAM */
     osimage    : ORIGIN = 0x00102000, LENGTH = 512K - 0x2000     /* Place where the main OS will end up */
     ram        : ORIGIN = 0x00200000, LENGTH = 64K - 0x20        /* RAM, minus small common area */
+    stack      : ORIGIN = 0x00200000 + 64K - 4K - 0x20, LENGTH = stacksize /* Stack */
```
Comment (doegox) [inline common_arm/ldscript.common:24]: shouldn't it be 
```
stack      : ORIGIN = 0x00200000 + 64K - stacksize - 0x20, LENGTH = stacksize
``` 
?

**Comment by doegox** [comment]
Sounds good to me. I pushed some small change

**Comment by iceman1001** [comment]
Great!
Let me think,  which functions are memory hungry?  SPIFFS?  json based standlone like HF_COLIN?

**Comment by iceman1001** [comment]
ok,   I have tested it.
hf 14a info -n
hf 14a reader
hf mf darkside
hf mf autopwd
hf mf fchk 1 m
sc info
hf iclass sim
hf iclass loclass
hf search
lf sniff
lf conf
lf hid sim
lf read
lf search
data plot 
its all happy.

**Comment by iceman1001** [comment]
@slurdge, did you add a line CHANGELOG.md ?   

Excellent changes bro!   🙏

**Pull request merged**